### PR TITLE
[YUNIKORN-135] Move the scheduler-interface related constants from th…

### PR DESCRIPTION
**Pull request by adamantal**
…e markdown to the common package

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -39,8 +42,14 @@ $(SI_PROTO).tmp: $(SI_SPEC)
 	(diff $@ $(SI_PROTO) > /dev/null 2>&1 || mv -f $@ $(SI_PROTO)) && \
 		rm -f $@
 
+$(CONSTANTS_GO): $(SI_SPEC)
```
Comment (wilfred-s) [inline Makefile:45]: In the proto spec build we force the build using a temp file. This guarantees a run of the build and then checks if the md document is still in sync with the generated proto file. This does not happen for the constants. Updates to the md document (i.e. a typo fix) could update and replace the constant.go file.
It also causes issues when I revert a change by checking out an older version of the md document. The build is not correctly replacing the generated `constants,go` file. Please setup an execution structure similar to the spec file with a move of a temp file etc:
```
CONSTANTS_GO := constants.go

$(CONSTANTS_GO).tmp: $(SI_SPEC)
	....
	(diff $@ $(PATH_CONSTANTS_GO) > /dev/null 2>&1 || mv -f $@ $(PATH_CONSTANTS_GO)) && \
		rm -f $@
```

File: lib/go/common/constants.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+const DefaultNodeAttributeRackNameKey = "si.io/rackname"
+
+// Application
+const LabelApp = "app"
+const LabelApplicationID = "applicationId"
+const LabelQueueName = "queue"
+
+// Resource
+const Memory = "memory"
+const CPU = "vcore"
```
Comment (wilfred-s) [inline lib/go/common/constants.go:36]: Resources are untyped in the core. I don't think we should limit types or set description in the SI for that reason.

File: scheduler-interface-spec.md [hunk 3]
```diff
@@ -730,32 +730,58 @@ message AllocationReleaseResponse {
 
 ### Following are constant of spec
 
-Scheduler Interface reserved all attribute in si.io namespace.
+Scheduler Interface reserved all attribute in si.io namespace, e.g. known attribute names for nodes and applications.
```
Comment (wilfred-s) [inline scheduler-interface-spec.md:733]: See the comment above, about `si.io`

File: lib/go/common/constants.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
+// **** DO NOT EDIT
+// **** This code is generated by the build process.
+
+package common
+
+// Constants for node attribtues
+const (
+	ARCH                = "si.io/arch"
```
[the remark this answers is not in the document]
Comment (adamantal) [reply lib/go/common/constants.go:8]: I'll modify the labels but I think it is going to be a breaking change, as `si.io` is hardcoded in the core in multiple places.

File: lib/go/common/constants.go (hunk 2 above)
Comment (adamantal) [reply lib/go/common/constants.go:36]: I don't really get what you mean here. The string constants "memory" and "vcore" are used frequently in the core as well. IMO it makes sense to move it to the SI and use them in both the shim and the core.

Comment (yangwwei) [reply lib/go/common/constants.go:32]: I think this is OK. @wilfred-s if you have more concerns, we can create another JIRA as a follow-up. At present, I think this is good to merge.

**Pull request merged**